app/utils/CheckYourAnswersHelper.scala

```diff
@@ -36,6 +37,66 @@ import utils.CheckYourAnswersHelper.dateFormatter
 
 class CheckYourAnswersHelper(userAnswers: UserAnswers)(implicit messages: Messages) {
 
+  def giveDetailsOfThisArrangement: Option[Row] = userAnswers.get(GiveDetailsOfThisArrangementPage) map {
```

Comment:
As a suggestion, the newly introduced rows could be separated in a trait and mixed up here.

Reply:
This is just from the template though we could change it as long as the whole dev team is on the same page